pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -171,6 +306,42 @@ def wrap_policy(*args, **kwargs):
 
         log.detail(f"{self.__class__.__name__}: exiting model_sharded_context.")
 
+    def determine_ddp_device_ids(self):
+        if self.root_device.type == "cpu":
+            return None
+        return [self.root_device.index]
+
+    def barrier(self, *args, **kwargs) -> None:
+        if not distributed_available():
+            return
+        if _TORCH_GREATER_EQUAL_1_8 and torch.distributed.get_backend() == "nccl":
+            torch.distributed.barrier(device_ids=self.determine_ddp_device_ids())
+        else:
+            torch.distributed.barrier()
+
+    def broadcast(self, obj: object, src: int = 0) -> object:
+        obj = [obj]
+        if self.global_rank != src:
+            obj = [None]
+        torch.distributed.broadcast_object_list(obj, src, group=_group.WORLD)
+        return obj[0]
```

Comment:
https://github.com/PyTorchLightning/pytorch-lightning/blob/79c4e5de60685dbc895641b0139ffc6180d069aa/pytorch_lightning/strategies/strategy.py#L37

```suggestion
    def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
        obj = [obj]
        if self.global_rank != src:
            obj = [None]
        torch.distributed.broadcast_object_list(obj, src, group=_group.WORLD)
        return obj[0]
```

Reply:
This is not necessarily true
- for pytorch version< 1.8, torch collective using broadcast, which returns Async work handle, if async_op is set to True. None, if not async_op or if not part of the group
- for pytorch version .=1.8 torch collective using broadcast, which returns None. If rank is part of the group, object_list will contain the broadcasted objects from src rank.

I think change obj to TBroadcast will cause issues